packages/test-utils/src/reporter.ts

```diff
@@ -42,14 +42,34 @@ export interface TestIdContext {
   attempt: number;
 }
 
+export type UploadStatusThreshold = "all" | "failed-and-flaky" | "failed";
+
+type UploadStatusThresholdInternal = UploadStatusThreshold | "none";
+
+export type UploadOption =
+  | boolean
+  | {
+      minimizeUploads?: boolean;
+      statusThreshold?: UploadStatusThreshold;
+    };
```

Comment:
nit: If we'll have to type check this in a few places, maybe it would be nice to make a type refinement util for it?

```ts
type AdvancedUploadOption = {
  minimizeUploads?: boolean;
  statusThreshold?: UploadStatusThreshold;
};

export type UploadOption =
  | boolean
  | AdvancedUploadOption;

function isAdvancedUploadOption(upload: UploadOption): upload is AdvancedUploadOption {
  return typeof config.upload === "object" && config != null;
}
```

Maybe not worth it if it's only in one spot though. It's just a habit I have.